File: oldapifile/docusign.java
/*
import com.docusign.esign.api.*;
import com.docusign.esign.client.*;
import com.docusign.esign.model.*;
import com.docusign.esign.client.auth.OAuth;
import com.docusign.esign.client.auth.OAuth.UserInfo;
import java.awt.Desktop;

import java.io.IOException;

import org.json.*;

public class DocuSignExample {
  public static void main(String[] args) {
    JSONObject obj = new JSONObject("credentials.json");
    String RedirectURI = obj.getJSONObject("Docusign_Java").getString("REDIRECT_URI");
    String ClientSecret =  obj.getJSONObject("Docusign_Java").getString("CLIENT_SECRET");
    String IntegratorKey = obj.getJSONObject("Docusign_Java").getString("INTEGRATOR_KEY");
    String BaseUrl = "https://demo.docusign.net/restapi";

    ApiClient apiClient = new ApiClient(BaseUrl);
    try
    {
        /////////////////////////////////////////////////////////////////////////////////////////////////////////
        // STEP 1: AUTHENTICATE TO RETRIEVE ACCOUNTID & BASEURL
        /////////////////////////////////////////////////////////////////////////////////////////////////////////

        String randomState = "*^.$DGj*)+}Jk";
        java.util.List<String> scopes = new java.util.ArrayList<String>();
        scopes.add(OAuth.Scope_SIGNATURE);
        // get DocuSign OAuth authorization url
        URI oauthLoginUrl = apiClient.getAuthorizationUri(IntegratorKey, scopes, RedirectURI, OAuth.CODE, randomState);
        // open DocuSign OAuth login in the browser
		    Desktop.getDesktop().browse(oauthLoginUrl);
        // IMPORTANT: after the login, DocuSign will send back a fresh
        // authorization code as a query param of the redirect URI.
        // You should set up a route that handles the redirect call to get
        // that code and pass it to token endpoint as shown in the next
        // lines:
        /*String code = "[once_you_get_the_oauth_code_put_it_here]";
        OAuth.OAuthToken oAuthToken = apiClient.generateAccessToken(IntegratorKey, ClientSecret, code);

        System.out.println("OAuthToken: " + oAuthToken);

        // now that the API client has an OAuth token, let's use it in all
        // DocuSign APIs
        UserInfo userInfo = apiClient.getUserInfo(oAuthToken.getAccessToken());

        System.out.println("UserInfo: " + userInfo);
        // parse first account's baseUrl
        // below code required for production, no effect in demo (same
        // domain)
        apiClient.setBasePath(userInfo.getAccounts().get(0).getBaseUri() + "/restapi");
        Configuration.setDefaultApiClient(apiClient);
		String accountId = userInfo.getAccounts().get(0).getAccountId();*/
		/*
          /////////////////////////////////////////////////////////////////////////////////////////////////////////
          // *** STEP 2: CREATE ENVELOPE FROM TEMPLATE
          /////////////////////////////////////////////////////////////////////////////////////////////////////////

          // create a new envelope to manage the signature request
          EnvelopeDefinition envDef = new EnvelopeDefinition();
          envDef.setEmailSubject("Docusign Photo Release Form");

          // assign template information including ID and role(s)
          envDef.setTemplateId(obj.getJSONObject("Docusign_Java").getString("TEMPLATE_ID"));

          // create a template role with a valid templateId and roleName and assign signer info
          TemplateRole tRole = new TemplateRole();
          tRole.setRoleName("Signer");
          tRole.setName(getName());
          tRole.setEmail(getEmail());

          // create a list of template roles and add our newly created role
          java.util.List<TemplateRole> templateRolesList = new java.util.ArrayList<TemplateRole>();
          templateRolesList.add(tRole);

          // assign template role(s) to the envelope
          envDef.setTemplateRoles(templateRolesList);

          // send the envelope by setting |status| to "sent". To save as a draft set to "created"
          envDef.setStatus("sent");

          // instantiate a new EnvelopesApi object
          EnvelopesApi envelopesApi = new EnvelopesApi(apiClient);

          // call the createEnvelope() API
          EnvelopeSummary envelopeSummary = envelopesApi.createEnvelope(accountId, envDef);
        }
        catch (ApiException ex)
        {
          System.out.println("Exception: " + ex);
        }
        catch (Exception e)
        {
          System.out.println("Exception: " + e.getLocalizedMessage());
        }
  }

  public String getName(){
    return name;
  }

  public String getEmail(){
    return email;
  }

  public void setName(String name){
      this.name = name;
  }

  public void setEmail(String email){
      this.email = email;
  }
}
*/

File: oldapifile/docusign.py
from __future__ import absolute_import, print_function
from pprint import pprint
import unittest
import webbrowser
from json import load

import docusign_esign as docusign
from docusign_esign import AuthenticationApi, TemplatesApi, EnvelopesApi
from docusign_esign.rest import ApiException

Class DocusignHelper():
    def __init__(self):
        load("credentials.json") as f:
            self._user_name = f["Docusign"]["Username"]
            self._integrator_key = f["Docusign"]["Integrator_Key"]
            self._base_url = "https://demo.docusign.net/restapi"
            self._oauth_base_url = "account.docusign.com"
            self._redirect_uri = "https://www.docusign.com/api"
            self._private_key_filename = "keys/docusign_private_key.txt"
            self._user_id = f["Docusign"]["User_ID"]
            self._template_id = f["Docusign"]["Template_ID"]
            self._api_client = docusign.ApiClient(base_url)
            self._oauth_login_url = api_client.get_jwt_uri(integrator_key, redirect_uri, oauth_base_url)

    def send_template(self, name, email):
        webbrowser.open_new_tab(self._oauth_login_url)
        print(self._oauth_login_url)

        self._api_client.configure_jwt_authorization_flow(
            self._private_key_filename, self._oauth_base_url, self._integrator_key, self._user_id, 3600)

        docusign.configuration.api_client = self._api_client

        template_role_name = 'Needs to sign'

        # create an envelope to be signed
        envelope_definition = docusign.EnvelopeDefinition()
        envelope_definition.email_subject = 'Please Sign my Python SDK Envelope'
        envelope_definition.email_blurb = 'Hello, Please sign my Python SDK Envelope.'

        # assign template information including ID and role(s)
        envelope_definition.template_id = self._template_id

        # create a template role with a valid template_id and role_name and assign signer info
        t_role = docusign.TemplateRole()
        t_role.role_name = template_role_name
        t_role.name = name
        t_role.email = email

        # create a list of template roles and add our newly created role
        # assign template role(s) to the envelope
        envelope_definition.template_roles = [t_role]

        # send the envelope by setting |status| to "sent". To save as a draft set to "created"
        envelope_definition.status = 'sent'

        auth_api = AuthenticationApi()
        envelopes_api = EnvelopesApi()

        try:
            login_info = auth_api.login(api_password='true', include_account_id_guid='true')
            assert login_info is not None
            assert len(login_info.login_accounts) > 0
            login_accounts = login_info.login_accounts
            assert login_accounts[0].account_id is not None

            self._base_url, _ = login_accounts[0].base_url.split('/v2')
            self._api_client.host = self._base_url
            docusign.configuration.api_client = api_client

            envelope_summary = envelopes_api.create_envelope(login_accounts[0].account_id, envelope_definition=envelope_definition)
            assert envelope_summary is not None
            assert envelope_summary.envelope_id is not None
            assert envelope_summary.status == 'sent'

            print("EnvelopeSummary: ", end="")
            pprint(envelope_summary)

        except ApiException as e:
            print("\nException when calling DocuSign API: %s" % e)
            assert e is None # make the test case fail in case of an API exception
